controlplane/kubeadm/internal/machine_collection.go

```diff
@@ -145,3 +152,13 @@ func (s FilterableMachineCollection) ConditionGetters() []conditions.Getter {
 	}
 	return res
 }
+
+// Names returns a slice of the names of each machine in the collection.
+// Useful for logging and test assertions.
+func (s FilterableMachineCollection) Names() []string {
```

Comment:
If the idea is to use this in tests, then there should probably be a test for this method to avoid bugs potentially affecting the test suite.